### PR TITLE
Added the filename to the error message when failed to load settings

### LINKED ISSUE #1061: Nunit30Settings.xml becomes corrupted
While trying to update a project with some failing tests (using NUnitForms), in some cases Nunit30Settings.xml becomes corrupted and is filled with a bunch of "nul"s, which, on the next run of nunit3-console results in the following exception:
```
     [exec] System.ApplicationException: Error loading settings file ---> System.Xml.XmlException: Root element is missing.
     [exec]    at System.Xml.XmlTextReaderImpl.ThrowWithoutLineInfo(String res)
     [exec]    at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
     [exec]    at System.Xml.XmlLoader.Load(XmlDocument doc, XmlReader reader, Boolean reserveWhitespace)
     [exec]    at System.Xml.XmlDocument.Load(XmlReader reader)
     [exec]    at System.Xml.XmlDocument.Load(String filename)
     [exec]    at NUnit.Engine.Internal.SettingsStore.LoadSettings()
     [exec]    --- End of inner exception stack trace ---
     [exec]    at NUnit.Engine.Internal.SettingsStore.LoadSettings()
     [exec]    at NUnit.Engine.Services.SettingsService.StartService()
     [exec]    at NUnit.Engine.Services.ServiceManager.StartServices()
     [exec]    at NUnit.Engine.TestEngine.Initialize()
     [exec]    at NUnit.Engine.TestEngine.NUnit.Engine.ITestEngine.get_Services()
     [exec]    at NUnit.ConsoleRunner.ConsoleRunner..ctor(ITestEngine engine, ConsoleOptions options, ExtendedTextWriter writer)
     [exec]    at NUnit.ConsoleRunner.Program.Main(String[] args)
```

As a workaround, one can delete this file, but it is pretty annoying and most importantly it wasn't clear what settings file it was referring to in the first place.

### PR DESCRIPTION
**Pull request by rprouse**
Fixes #1061

### CODE REVIEW

File: src/NUnitEngine/nunit.engine/Internal/SettingsStore.cs [hunk 1]
```diff
@@ -89,7 +89,8 @@ public void LoadSettings()
             }
             catch (Exception ex)
             {
-                throw new ApplicationException("Error loading settings file", ex);
+                string msg = string.Format("Error loading settings {0}.", _settingsFile);
+                throw new ApplicationException(msg, ex);
```
Comment (CharliePoole) [inline src/NUnitEngine/nunit.engine/Internal/SettingsStore.cs]: I think this should be an NUnitEngineException, which will get displayed without the stack trace.

**Comment by CharliePoole** [comment]
See my note about the exception type. Otherwise, it's ready to merge.

File: src/NUnitEngine/nunit.engine/Internal/SettingsStore.cs (hunk 1 above)
Comment (rprouse) [inline src/NUnitEngine/nunit.engine/Internal/SettingsStore.cs]: Good idea, I will update. Do we show the inner exception in that case? If not, I will add the message from the inner exception. When I was testing, it told me exactly what was wrong with the file.

**Comment by rprouse** [comment]
Changes based on your feedback have been pushed. The message from the inner exception is not displayed with the NUnitEngineException, so I added it to the message to tell people what is wrong with the settings file.

**Pull request merged**